src/components/Layout/GridTableLayout/GridTableLayout.tsx

```diff
@@ -89,14 +99,43 @@ function GridTableLayoutComponent<
   X extends Only<GridTableXss, X>,
   QData,
 >(props: GridTableLayoutProps<F, R, X, QData>) {
-  const { pageTitle, breadcrumb, tableProps, layoutState, primaryAction, secondaryAction, tertiaryAction } = props;
+  const {
+    pageTitle,
+    breadcrumb,
+    tableProps,
+    layoutState,
+    primaryAction,
+    secondaryAction,
+    tertiaryAction,
+    hideEditColumns = false,
+  } = props;
 
+  const tid = useTestIds(props);
+  const columns = tableProps.columns;
+
+  !hideEditColumns && validateColumns(columns);
+
+  const hasHideableColumns = useMemo(() => columns.some((c) => c.canHide), [columns]);
```

Comment:
Ironically I would probably have not `useMemo-d` this `columns.some` (it seems cheap), but would `useMemo` the `validateColumns` (albeit I think it is also cheap), but maybe we could blend both by keeping the `useMemo` here, but do the `validateColumns` inside of a lambda here, so we only do it when columns change.